### PR TITLE
Parsing for REFRESH's automated cluster scheduling

### PR DESCRIPTION
**Pull request by ggevay**
This PR just adds the 
`SCHEDULE = {MANUAL | ON REFRESH}`
option to `CREATE CLUSTER` and `ALTER CLUSTER`.
`REFRESH` is not yet implemented, so we just error if the user tries to specify it.

Discussion on whether to have `REFRESH` or `ON REFRESH` [here](https://materializeinc.slack.com/archives/C063H5S7NKE/p1710355545343079).

### Motivation

  * This PR is the first step towards adding a known-desirable feature: https://github.com/MaterializeInc/materialize/issues/25712

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Comment by ggevay** [comment]
Rebased and resolved some merge conflicts.

@ParkMyCar, could you please review this PR?

File: src/sql-parser/src/ast/defs/statement.rs [hunk 1]
```diff
@@ -3579,6 +3586,25 @@ impl<T: AstInfo> AstDisplay for RefreshOptionValue<T> {
...
+}
+
+impl AstDisplay for ClusterScheduleOptionValue {
+    fn fmt<W: fmt::Write>(&self, f: &mut AstFormatter<W>) {
+        match self {
+            ClusterScheduleOptionValue::Manual => {
+                f.write_str("MANUAL");
+            }
+            ClusterScheduleOptionValue::Refresh => {
+                f.write_str("REFRESH");
```
Comment (ParkMyCar) [inline src/sql-parser/src/ast/defs/statement.rs:3602]: Shouldn't this be `ON REFRESH`? I think as-is the create_sql won't roundtrip

File: src/sql-parser/tests/testdata/ddl [hunk 2]
```diff
@@ -1508,14 +1508,14 @@ CreateCluster(CreateClusterStatement { name: Ident("cluster"), options: [Cluster
 parse-statement
```
Comment (ParkMyCar) [inline src/sql-parser/tests/testdata/ddl:1508]: Can you add a test to this file with the new syntax to make sure everything roundtrips?

**Review by ParkMyCar — CHANGES_REQUESTED** [review]
Wooo! Excited for this! Just need the one change really and we should be good

File: src/sql-parser/src/ast/defs/statement.rs (hunk 1 above)
Comment (ggevay) [reply src/sql-parser/src/ast/defs/statement.rs]: Oops, sorry, fixed!

File: src/sql-parser/tests/testdata/ddl (hunk 2 above)
Comment (ggevay) [reply src/sql-parser/tests/testdata/ddl:1508]: Absolutely! Done.

**Review by ParkMyCar — APPROVED** [review]
Thanks!

**Pull request merged**